src/mds/MDCache.cc

```diff
@@ -13557,6 +13557,13 @@ void MDCache::handle_mdsmap(const MDSMap &mdsmap, const MDSMap &oldmap) {
   }
 }
 
+bool MDCache::is_ready_to_trim_cache(void)
+{
+  // null rejoin_done means rejoin has finished and all the rejoin acks
+  // have been well received.
+  return is_open() && !rejoin_done;
+}
```

Comment:
I'm a bit confused here, the comment in the above code says that `rejoin_done` will be null when all the rejoin acks have been received i.e. the last cache_rejoin ack must have been received. but what i understand from `return is_open() && !rejoin_done;` is that "trim only if the session is open and rejoin_done is not null i.e not all rejoin acks are received", is this correct interpretation? What i feel from the tracker and the PR title is we should wait till all the acks are done right? i.e. wait till the `rejoin_done` is null so shouldn't the code be `return is_open() && rejoin_done`?

Reply:
As we discussed, when `return is_open() && !rejoin_done;` is true it will be ready to trim.